crates/amaru-kernel/src/lib.rs

```diff
@@ -485,15 +485,18 @@ pub fn encode_bech32(hrp: &str, payload: &[u8]) -> Result<String, Box<dyn std::e
 
 /// Calculate the epoch number corresponding to a given slot on the PreProd network.
 // TODO: Design and implement a proper abstraction for slot arithmetic. See https://github.com/pragma-org/amaru/pull/26/files#r1807394364
+#[allow(clippy::unwrap_used)]
 pub fn epoch_from_slot(slot: u64) -> u64 {
-    let shelley_slots = slot - BYRON_TOTAL_SLOTS as u64;
-    (shelley_slots / SHELLEY_EPOCH_LENGTH as u64) + PREPROD_SHELLEY_TRANSITION_EPOCH as u64
+    // FIXME: currently hardwired to preprod network
+    preprod_era_history().slot_to_epoch(slot).unwrap()
```

Comment:
For completeness, would it make sense to also adjust `relative_slot` and `next_epoch_first_slot` ?

Reply:
Also, can it realistically fail? Do we want to propagate the error?

Reply:
It can if we ever request a slot before the start of the shelley era; Which can totally happen is someone submits a transaction with such a lower bound.

Reply:
or a slot past the `EraHistory` horizon, which is the reason for the `unwrap()`

Reply:
Then I think it's worth bubbling the error up. `unwrap` should only be used when there is no way it can happen for soundness reason, and that somehow the type system doesn't help with expressing this.

Reply:
Yes, that's what I am working on now